.github/workflows/pipeline.yml

```diff
@@ -63,57 +75,99 @@ jobs:
+        run: esy cleanup .
 
-      - name: Get Esy UsePpx binary path
-        id: esy_use_ppx_binary_path
-        shell: bash
+  prepare-publish:
+    name: Prepare publish to npm
+    needs: build_and_test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/setup-node@v1
+        with:
+          node-version: 12
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 1
+          ref: ${{ github.ref }}
+          submodules: true
+      # Download platform artifacts
+      - name: Download Linux release
+        uses: actions/download-artifact@v2
+        with:
+          name: ubuntu-latest
+          path: ubuntu-latest
+      - name: Download Windows release
+        uses: actions/download-artifact@v2
+        with:
+          name: windows-latest
+          path: windows-latest
+      - name: Download macOS release
+        uses: actions/download-artifact@v2
+        with:
+          name: macos-latest
+          path: macos-latest
+      # Move artifacts in place
+      - name: Move artifacts
         run: |
-          # from https://stackoverflow.com/a/24848739/617787
-          s=$(pwd); d=$(dirname $(esy x which use-tailwind-ppx));
-          while [ "${d#$s/}" == "${d}" ]
-          do s=$(dirname $s);b="../${b}"; done;
-          if [ "${{ matrix.os }}" == "windows-latest" ]; then
-            ESY__BINARY_PATH=${b}${d#$s/}/use-tailwind-ppx.exe
-          else
-            ESY__BINARY_PATH=${b}${d#$s/}/use-tailwind-ppx
-          fi
-          echo "ESY__BINARY_PATH: $ESY__BINARY_PATH"
-          echo "##[set-output name=path;]$ESY__BINARY_PATH"
-
-      - name: Upload use-ppx artifacts for ${{ matrix.os }}
-        if: github.event_name != 'pull_request' && startsWith(github.ref, 'refs/tags/')
-        uses: actions/upload-artifact@master
+          mkdir -p _release/platform-linux
+          mkdir -p _release/platform-windows-x64
+          mkdir -p _release/platform-darwin
+          cp -a ubuntu-latest/. _release/platform-linux
+          cp -a windows-latest/. _release/platform-windows-x64
+          cp -a macos-latest/. _release/platform-darwin
+      - name: Prepare package
+        run: node .github/workflows/bundle-release.js
+      # Create a npm package that can easily be published and tested
+      - name: npm pack
+        run: npm pack .
+        working-directory: _release
+      - name: move package
+        run: mv _release/*.tgz tailwind-ppx.tgz
+      # Upload artifacts
+      - uses: actions/upload-artifact@v2
         with:
-          name: ${{ matrix.os }}
-          path: ${{ steps.esy_use_ppx_binary_path.outputs.path }}
+          name: release
+          path: _release/
+      - uses: actions/upload-artifact@v2
+        with:
+          name: release-tarball
+          path: tailwind-ppx.tgz
 
+  test-platform:
+    name: Test installing ${{ matrix.os }}
+    needs: prepare-publish
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
```

Comment:
Same question as above

Reply:
Same idea, so we always complete the step on all OS and don't stop as soon as it fails on one.